### DISCUSSION TITLE
Problems Accessibility

### DISCUSSION THREAD

**Discussion by torresgol10** [orestbida/cookieconsent#575] [General]
I have a problem of accessibility in my web, because if a user is filling a form and gives a tab, when he has not accepted the cookie message, the focus moves to the cookie message.

pls, help.

**Comment by TonyMonast** [comment]
Hi,

The author of the plugin is better placed than me to answer you, but here are my observations.

If you use force_consent with true, it is best to leave this behavior as it is.

What forces the TAB to be done in the dialog box is the [line 1509](https://github.com/orestbida/cookieconsent/blob/43a1d2ba9c93543b7dd2d229bb56194808ad3247/src/cookieconsent.js#L1509)

```
// Accessibility :=> if tab pressed => trap focus inside modal
setTimeout(function(){_handleFocusTrap();}, 100);
```

I changed the code for
```
if(_config.force_consent) {
    // Accessibility :=> if tab pressed => trap focus inside modal
    setTimeout(function(){_handleFocusTrap();}, 100);
}
```

I also changed the line [Line 234](https://github.com/orestbida/cookieconsent/blob/43a1d2ba9c93543b7dd2d229bb56194808ad3247/src/cookieconsent.js#L234) for the function _setConfig because the reference to force_consent was missing. I added this :

```
if(typeof user_config['force_consent'] === "boolean")
    _config.force_consent = user_config['force_consent'];
```

From then on if force_consent is true, the TAB focuses on the dialog, otherwise the TAB behaves normally.

However, from an accessibility point of view, it may still be relevant that TAB always focus, force_consent or not, on the dialog box so that the user navigating with keyboard can quickly respond by doing TAB, rather than browsing the entire document to get there.

**Comment by orestbida** [comment]
Hi @torresgol10, which version are you using? This will be fixed in the next v3 release.

**Reply by torresgol10** [reply]
Perfect, we will use the new version, thank you.
